File: BeatSync-Service-Persistence/src/main/java/com/astimefades/beatsyncservice/model/response/Response.java
package com.astimefades.beatsyncservice.model.response;

import com.fasterxml.jackson.annotation.JsonProperty;

public class Response<T> {

    private T payload;
    private int errorNumber;
    private String errorDescription;

    public Response() {}

    public Response(T payload) {
        this.payload = payload;
    }

    @JsonProperty("payload")
    public T getPayload() {
        return payload;
    }

    public void setPayload(T payload) {
        this.payload = payload;
    }

    @JsonProperty("errorNumber")
    public int getErrorNumber() {
        return errorNumber;
    }

    public void setErrorNumber(int errorNumber) {
        this.errorNumber = errorNumber;
    }

    @JsonProperty("errorDescription")
    public String getErrorDescription() {
        return errorDescription;
    }

    public void setErrorDescription(String errorDescription) {
        this.errorDescription = errorDescription;
    }
}

File: BeatSync-Client-Android/app/src/main/java/com/astimefades/beatsyncandroid/model/response/Response.kt
package com.astimefades.beatsyncandroid.model.response

import com.fasterxml.jackson.annotation.JsonProperty

/**
 * Created by tobycaulk on 2/27/18.
 */
data class Response<T>(@JsonProperty("payload") val payload: T,
                       @JsonProperty("errorNumber") val errorNumber: Int??,
                       @JsonProperty("errorDescription") val errorDescription: String??)